tests/std/tests/P0475R1_P0591R4_uses_allocator_construction/test.cpp

```diff
@@ -0,0 +1,181 @@
+    }
+
+    { // pair() overload
+        auto tuple6 =
+            uses_allocator_construction_args<pair<DefaultConstructible, OnlyAllocatorArgConstructible>>(alloc);
+        static_assert(is_same_v<decltype(tuple6),
+            tuple<piecewise_construct_t, DefaultConstructArgs, OnlyAllocatorArgConstructArgs>>);
+
+        auto tuple7 = uses_allocator_construction_args<pair<OnlyAllocatorConstructible, DefaultConstructible>>(alloc);
+        static_assert(is_same_v<decltype(tuple7),
+            tuple<piecewise_construct_t, OnlyAllocatorConstructArgs, DefaultConstructArgs>>);
+    }
+
+    { // pair(first, second) overload
+        auto tuple4 = uses_allocator_construction_args<pair<int, AllocatorArgConstructible>>(alloc, i, i);
+        static_assert(
+            is_same_v<decltype(tuple4), tuple<piecewise_construct_t, tuple<int&>, AllocatorArgConstructArgs>>);
+
+        auto tuple5 = uses_allocator_construction_args<pair<AllocatorConstructible, int>>(alloc, i, i);
+        static_assert(is_same_v<decltype(tuple5), tuple<piecewise_construct_t, AllocatorConstructArgs, tuple<int&>>>);
+    }
+
+    { // pair(const pair&) overload
+        auto tuple4 = uses_allocator_construction_args<pair<int, AllocatorArgConstructible>>(alloc, p);
+        static_assert(is_same_v<decltype(tuple4),
+            tuple<piecewise_construct_t, tuple<const int&>, ConstAllocatorArgConstructArgs>>);
+
+        auto tuple5 = uses_allocator_construction_args<pair<AllocatorConstructible, int>>(alloc, p);
+        static_assert(
+            is_same_v<decltype(tuple5), tuple<piecewise_construct_t, ConstAllocatorConstructArgs, tuple<const int&>>>);
+    }
+
+    { // pair(pair&&) overload
+        auto tuple4 = uses_allocator_construction_args<pair<int, AllocatorArgConstructible>>(alloc, move(p));
+        static_assert(
+            is_same_v<decltype(tuple4), tuple<piecewise_construct_t, tuple<int&&>, MovedAllocatorArgConstructArgs>>);
+
+        auto tuple5 = uses_allocator_construction_args<pair<AllocatorConstructible, int>>(alloc, move(p));
+        static_assert(
+            is_same_v<decltype(tuple5), tuple<piecewise_construct_t, MovedAllocatorConstructArgs, tuple<int&&>>>);
+    }
+
+    {
+        auto obj1 = make_obj_using_allocator<AllocatorArgConstructible>(alloc, i);
+        static_assert(is_same_v<decltype(obj1), AllocatorArgConstructible>);
+        assert(obj1.x == i);
+
+        auto obj2 = make_obj_using_allocator<AllocatorConstructible>(alloc, i);
+        static_assert(is_same_v<decltype(obj2), AllocatorConstructible>);
+        assert(obj2.x == i);
+    }
+
+    {
+        allocator<AllocatorArgConstructible> alloc2;
+        auto ptr1 = alloc2.allocate(1);
+
+        uninitialized_construct_using_allocator(ptr1, alloc, i);
+        assert(ptr1->x == i);
+        destroy_at(ptr1);
+
+        alloc2.deallocate(ptr1, 1);
+
+        allocator<AllocatorConstructible> alloc3;
+        auto ptr2 = alloc3.allocate(1);

```

Comment:
Naming nitpick: It's slightly confusing that `ptr1` is what `alloc2` allocates, and `ptr2` is what `alloc3` allocates. Matching numbers would be more readable.